pyartifactory/models/artifact.py

```diff
@@ -10,28 +10,33 @@
 
 from pydantic import BaseModel
 
+from pyartifactory.exception import InvalidAlgorithmError
+
 
 class Checksums(BaseModel):
     """Models a checksum."""
 
-    sha1: str
-    md5: str
-    sha256: str
+    sha1: Optional[str] = None
+    md5: Optional[str] = None
+    sha256: Optional[str] = None
 
     @classmethod
-    def generate(cls, file_: Path) -> Checksums:
+    def generate(cls, file_: Path, algorithms: List[str]) -> Checksums:
         block_size: int = 65536
         mapping: dict[str, Callable[[], Any]] = {"md5": hashlib.md5, "sha1": hashlib.sha1, "sha256": hashlib.sha256}
         results = {}
 
-        for algorithm, hashing_function in mapping.items():
-            hasher = hashing_function()
-            with file_.absolute().open("rb") as fd:
-                buf = fd.read(block_size)
-                while len(buf) > 0:
-                    hasher.update(buf)
+        for algorithm in algorithms:
```

Comment:
In my opinion, the old generate checksum works fine and does not need to be updated. Here you want to pass as input a list of algorithms, why that? What would happen if a user forget an algorithm (Sha256 for example) ? Will we deploy only 2 checksums ?

Reply:
I did it that way thinking of not everyone wants to be punished on performance for calculating all the checksums. 
`Checksums.generate` is an internal class, in case a user forget an algoritm, the default ones will be used. You can see that at https://github.com/anancarv/python-artifactory/pull/199/files/cdd908f6a1dcd63114628f5beac86e654e6a88ea#diff-bd8c98fdc28a0b44ac12a172a61699adfe00b9254b02255b2b33b567ba2ba621R106

Reply:
I think it's preferable to keep it simple here. Also, I don't see use cases where people want to calculate only some checksums. Let's wait for more people to request that feature instead of unnecessarily complexifying the codebase